tests/performance-tests/src/utils/publicApiService.ts

```diff
@@ -0,0 +1,290 @@
+  and: DataSetQueryNode[];
+}
+
+export interface DataSetQueryConditionOr {
+  or: DataSetQueryNode[];
+}
+
+export interface DataSetQueryConditionNot {
+  not: DataSetQueryNode[];
+}
+
+export type DataSetQueryNode =
+  | DataSetQueryConditionAnd
+  | DataSetQueryConditionOr
+  | DataSetQueryConditionNot
+  | DataSetQueryCriteria;
+
+interface TimePeriod {
+  code: string;
+  year: number;
+}
+
+export interface DataSetQueryRequest {
+  facets: DataSetQueryNode;
+  indicators?: string[];
+  sort?: [
+    {
+      name: string;
+      order: string;
+    },
+  ];
+}
+
+export interface DataSetQueryResponse {
+  footnotes: [
+    {
+      id: string;
+      content: string;
+    },
+  ];
+  paging: {
+    page: 0;
+    pageSize: 0;
+    totalResults: 0;
+    totalPages: 0;
+  };
+  results: {
+    id: string;
+  }[];
+}
+
+// Some Publications and their data sets are currently not discoverable via
+// the API, so we can include them manually here.
+const hiddenPublications: Publication[] = [
+  {
+    id: '1681557f-510f-446e-bc9a-f2c7a59d1cfa',
+    slug: 'benchmark-publication',
+    title: 'Benchmark Publication',
+  },
+];
+
+const hiddenDataSets: { [publicationId: string]: { dataSets: DataSet[] } } = {
+  '1681557f-510f-446e-bc9a-f2c7a59d1cfa': {
+    dataSets: [
+      {
+        id: 'a96044e5-2310-4890-a601-8ca0b67d2964',
+        name: 'QUA01',
+        content: '',
+        geographicLevels: ['National'],
+        timePeriods: {
+          start: '2013/14',
+          end: '2018/19',
+        },
+      },
+    ],
+  },
+};
+
+class PublicApiService {
+  private readonly client: HttpClient;
+
+  constructor(baseUrl: string, checkResponseStatus = true) {
+    this.client = new HttpClient({
+      baseUrl,
+      checkResponseStatus,
+    });
+  }
+
+  listPublications() {
+    const { response, json } = this.client.get<Paged<Publication>>(
+      '/v1/publications?pageSize=40',
+    );
+    const publications: Publication[] = [
+      ...json.results,
+      ...hiddenPublications,
+    ];
+    return {
+      publications,
+      response,
+    };
+  }
+
+  listDataSets(publicationId: string) {
+    let response: RefinedResponse<'text'> | undefined;
+    let json: DataSet[];
+
+    if (Object.keys(hiddenDataSets).includes(publicationId)) {
+      json = hiddenDataSets[publicationId].dataSets;
+    } else {
+      const result = this.client.get<DataSet[]>(
+        `/v1/publications/${publicationId}/data-sets`,
+      );
+      response = result.response;
+      json = result.json;
+    }
+    return {
+      dataSets: json,
+      response,
+    };
+  }
+
+  getDataSetsMeta(dataSetId: string) {
+    const { response, json } = this.client.get<DataSetMeta>(
+      `/v1/data-sets/${dataSetId}/meta`,
+    );
+    return {
+      meta: json,
+      response,
+    };
+  }
+
+  getDataSetQueryUrl({ dataSetId, page }: { dataSetId: string; page: number }) {
+    return `/v1/data-sets/${dataSetId}/query?page=${page}&pageSize=10000`;
```

Comment:
Mentioned in Slack that it would be a good idea to make the `pageSize` configurable so we can test for a good max pagination limit. The current 10000 was a bit arbitrarily set and definitely introduces potential performance issues due to its size.

We'll want to do some more testing to try and figure out a better limit.

Reply:
Will leave this as is for now and suggest that Nusrath or I look to make this configurable when we get a chance to do EES-4190.